src/index.ts

```diff
@@ -77,8 +77,39 @@ export interface Options {
     offsetFrom?: string
 }
 
+/**
+ * Provides access to the data within a Record that was consumed
+ */
+export interface Record {
+    /**
+     * Returns the Key of this Record as a string, or null if there is no key
+     */
+    key(): string | null
+
+    /**
+     * Returns true if this Record has a key, or false otherwise
+     */
+    hasKey(): boolean
+
+    /**
+     * Returns the Value of this Record as a string
+     */
+    value(): string
+
+    /**
+     * Returns the Key of this Record as a byte buffer, or null if there is no key
+     */
+    keyBuffer(): ArrayBuffer | null
+
+    /**
+     * Returns the Value of this Record as a byte buffer
+     */
+    valueBuffer(): ArrayBuffer
+}
+
 export interface TopicProducer {
     sendRecord(data: string, partition: number): Promise<void>
+    send(key: string, value: string): Promise<void>
```

Comment:
ditto. see comment about key

Reply:
I think that for the producer, there should be some kind of convenience method that lets users send strings. In Rust, we have conversion traits that help us out with this, e.g.

```rust
pub fn send<K: AsRef<[u8]>, V: AsRef<[u8]>>(&self, key: K, value: V) { ... }

// we can pass anything that can be referenced as bytes, including String, Vec<u8>, &str, &[u8]...
producer.send("hello", "world")
producer.send(vec![1,2,3], vec![4,5,6])
// ... whatever else
```

I don't think we will be able to do something quite so clean in JS, but I think it might be good to include at least something like

```typescript
sendString(key: string, value: string)
```

One thing I don't like about this is the similarity to `sendRecord`. It makes it seem like a record is one kind of thing you would send, and a string is a different kind of thing you would send. I am open to talking about other potential options, but if possible, I'd like to come up with a decent solution for strings.

Reply:
Managed to get this to work:

```typescript
async send(key: string | ArrayBuffer, value: string | ArrayBuffer)
```